.github/workflows/release.yaml

```diff
@@ -85,151 +203,199 @@ jobs:
-
-    - name: Set NUCLIO_LABEL to release tag
-      if: github.event_name == 'release'
-      run: echo "NUCLIO_LABEL=${{ steps.release_info.outputs.REF_TAG }}" >> $GITHUB_ENV
-
-    - uses: actions/checkout@v4
-
-    - uses: actions/setup-go@v5
-      with:
-        cache: true
-
-    - name: Install QEMU
-      if: matrix.arch != 'amd64'
-      uses: docker/setup-qemu-action@v3
-      with:
-        platforms: ${{ matrix.arch }}
-
-    - name: Build binaries
-      run: |
-        NUCLIO_OS=linux make tools
-        NUCLIO_OS=darwin make tools
-        if [ $NUCLIO_ARCH == "amd64" ]; then \
-          NUCLIO_OS=windows make tools; \
-        fi;
-      env:
-        NUCLIO_NUCTL_CREATE_SYMLINK: false
-        GOPATH: /home/runner/go
-        DOCKER_DEFAULT_PLATFORM: linux/${{ matrix.arch }}
-
-    - name: Upload binaries
-      uses: AButler/upload-release-assets@v3.0
-      with:
-        files: '/home/runner/go/bin/nuctl-*'
-        repo-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Prepare envs
+        run: |
+          echo "NUCLIO_ARCH=${{ matrix.arch }}" >> $GITHUB_ENV
+
+      - name: Prepare outputs
+        id: release_info
+        run: |
+          echo "REF_BRANCH=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
+          echo "REF_TAG=${GITHUB_REF#refs/tags/}" >> $GITHUB_OUTPUT
+
+      - name: Set NUCLIO_LABEL to unstable
+        if: github.event_name == 'push'
+        run: echo "NUCLIO_LABEL=unstable" >> $GITHUB_ENV
+
+      - name: Set NUCLIO_LABEL to release tag
+        if: github.event_name == 'release'
+        run: echo "NUCLIO_LABEL=${{ steps.release_info.outputs.REF_TAG }}" >> $GITHUB_ENV
+
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-go@v5
+        with:
+          cache: true
+
+      - name: Install QEMU
+        if: matrix.arch != 'amd64'
+        uses: docker/setup-qemu-action@v3
+        with:
+          platforms: ${{ matrix.arch }}
+
+      - name: Build binaries
+        run: |
+          NUCLIO_OS=linux make tools
+          NUCLIO_OS=darwin make tools
+          if [ $NUCLIO_ARCH == "amd64" ]; then \
+            NUCLIO_OS=windows make tools; \
+          fi;
+        env:
+          NUCLIO_NUCTL_CREATE_SYMLINK: false
+          GOPATH: /home/runner/go
+          DOCKER_DEFAULT_PLATFORM: linux/${{ matrix.arch }}
+
+      - name: Upload binaries
+        uses: AButler/upload-release-assets@v3.0
+        with:
+          files: '/home/runner/go/bin/nuctl-*'
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+  helm-publish:
+    name: Publish helm charts
+    if: github.event_name == 'workflow_dispatch'
+    needs: [ release-images ]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+         ref: ${{ github.event.inputs.branch }}
+
+      - name: Checkout gh-pages branch
+        uses: actions/checkout@v4
+        with:
+          ref: gh-pages
+          path: nuclio-helm
```

Comment:
why is `path` needed ?

Reply:
@TomerShor gh-pages branch will be kept in this path. Since we release charts from the specified branch, we need to be checkout-ed at this branch, and  at the same time we need to modify gh-pages branch. This param allows working with many branches at once